integr8ly/integreatly-operator
# MGDAPI-3878 - Added annotation to 3scale api-manager to skip apicast service port reconcile

# Issue link
[MGDAPI-3878](https://issues.redhat.com/browse/MGDAPI-3878) & [THREESCALE-8385](https://issues.redhat.com/browse/THREESCALE-8385)
# What
Added function in 3scale reconciler to reconcile annotation to allow rate limit to take over apicast service port configuration
# Verification steps
- Change the 3scale image in the CSV to `quay.io/bkeane0/3scale-operator:0.4.0` (my image from the related 3scale PR to skip service port reconcile) and install RHOAM
- You'll also need to implement the rules from this PR https://github.com/3scale/3scale-operator/commit/d56aaee25608988641e16cf92b332c886948523d into integreatly for testing purposes.
- Update the APIManager CRD to match the latest 3scale version
- There may be some issue with 3scale install / the CRD for the apimanager so give 3scale one reconcile and then add the `externalComponents` section from here https://github.com/3scale/3scale-operator/blob/d4b44f6d04398b5858659729c899304b8dc3ff8f/bundle/manifests/apps.3scale.net_apimanagers.yaml#L2598-L2617 to the CRD manually
- Check that the `APIManager` CR has the annotation `apps.3scale.net/disable-apicast-service-reconciler: 'true'` after you have fixed this problem with the CRD (I've also run into issues with `FILE_UPLOAD_STORAGE` being duplicated and stopping 3scale install but this is a separate issue
- Check the `apicast-production` and `apicast-staging` services that the `gateway` port is set to `8443`
- Make a change to either apicast `deploymentConfig` to trigger the apicast reconciler.
- Watch both services and ensure the gateway port does not change to `8080`
